diff -aur fam-2.7.0.orig/src/IMon.c++ fam-2.7.0/src/IMon.c++
--- fam-2.7.0.orig/src/IMon.c++	2003-01-18 15:18:12.000000000 +0100
+++ fam-2.7.0/src/IMon.c++	2008-03-20 23:30:25.000000000 +0100
@@ -40,7 +40,7 @@
 #include "Interest.h"
 #include "Log.h"
 #include "Scheduler.h"
-#include "alloc.h"
+#include <memory>
 
 int		   IMon::imonfd = -2;
 IMon::EventHandler IMon::ehandler = NULL;
Only in fam-2.7.0/src: IMon.c++~
