diff -aur fam-2.7.0.orig/src/DNotify.c++ fam-2.7.0/src/DNotify.c++
--- fam-2.7.0.orig/src/DNotify.c++	2008-03-20 23:31:08.000000000 +0100
+++ fam-2.7.0/src/DNotify.c++	2008-03-20 23:38:28.000000000 +0100
@@ -41,7 +41,7 @@
 #include "Interest.h"
 #include "Log.h"
 #include "Scheduler.h"
-#include "alloc.h"
+#include <memory>
 
 
 int DNotify::pipe_write_fd = -2;
Only in fam-2.7.0/src: DNotify.c++~
